### PR TITLE
2021.9.x feature gigs

### PR DESCRIPTION
**Pull request by tjerman**
wip

### CODE REVIEW

File: pkg/gig/source_file.go [hunk 1]
```diff
@@ -0,0 +1,318 @@
...
+
+func (f *fileSource) SetDecoders(dd ...DecodeTask) {
+	f.decoders = dd
+}
+
+func (f fileSource) Decoders() []DecodeTask {
+	return f.decoders
+}
+
+func (f fileSource) NasDecoders() bool {
```
Comment (darh) [inline pkg/gig/source_file.go]: typo? HasDecoders?

File: pkg/gig/service.go [hunk 2]
```diff
@@ -0,0 +1,302 @@
...
+func (svc *service) Create(ctx context.Context, w Worker) (g *Gig, err error) {
+	return updateGig(ctx, newGig(w))
+}
+
+// Read returns the requested gig
+func (svc *service) Read(ctx context.Context, id uint64) (*Gig, error) {
+	return getGig(ctx, id)
+}
+
+func (svc *service) AddSources(ctx context.Context, gigID uint64, sources SourceSet, decoders ...DecodeTask) (g *Gig, err error) {
```
Comment (darh) [inline pkg/gig/service.go]: Why do you want to keep these gig-modifying functions on the service and not on the gig struct (or as a standalone functions)? Where do we benefit if we drive all this changes through gig service?

File: pkg/gig/task.go [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+		Decode(context.Context, Source) (SourceSet, error)
+	}
+
+	PreprocessTask interface {
+		Ref() string
+		Worker() []string
+		Params() interface{}
+	}
+
+	PostprocessTask interface {
```
Comment (darh) [inline pkg/gig/task.go]: "Postprocessor"

File: pkg/gig/task.go [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+	}
+	PostprocessTaskWrapSet []*PostprocessTaskWrap
+
+	DecodeTask interface {
+		Source() uint64
+		CanDecode(Source) bool
+		Decode(context.Context, Source) (SourceSet, error)
+	}
+
+	PreprocessTask interface {
```
Comment (darh) [inline pkg/gig/task.go]: "Preprocessor"

File: pkg/gig/task.go [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+	}
+	PreprocessTaskWrapSet []*PreprocessTaskWrap
+
+	PostprocessTaskWrap struct {
+		Ref    string
+		Params interface{}
+	}
+	PostprocessTaskWrapSet []*PostprocessTaskWrap
+
+	DecodeTask interface {
```
Comment (darh) [inline pkg/gig/task.go]: Decoder

File: pkg/gig/service.go (hunk 2 above)
Comment (tjerman) [reply pkg/gig/service.go]: Discussed

**Comment by tjerman** [comment]
Moving...

**Pull request closed without merging**